app/src/main/kotlin/com/wire/android/ui/userprofile/UserProfileState.kt

```diff
@@ -8,7 +10,55 @@ data class UserProfileState(
     val fullName: String,
     val userName: String,
     val teamName: String, //maybe teamId is better here
-    val otherAccounts: List<OtherAccount>
+    val otherAccounts: List<OtherAccount>,
+    val dialogState: DialogState = DialogState.None
 )
 
 data class OtherAccount(val id: String, val avatarUrl: String, val fullName: String, val teamName: String? = null)
+
+sealed class DialogState {
+
+    object None : DialogState()
+
+    sealed class StatusChange(
+        @StringRes val title: Int,
+        @StringRes val text: Int,
+        val status: UserStatus,
+    ) : DialogState() {
```

Comment:
Shouldn't we make `None` an equivalent dialog state as the rest (Available, Busy, Away)? That way there is no need to have nested sealed classes and 2 different `None` and `StateNone` types of dialog.

Perhaps we could make `DialogStatusInfo` an interface that can be implemented by the different states.

Reply:
Imho this class should be dealing only with the specific User Status Availability dialogs. If you want to reuse this dialog class, it would be better for it to live in a different UI folder focused on dialogs or UI utils, and that can be generic enough for other components to reuse it. Then we will make all these `DialogState` objects/classes inherit or implement it.  Otherwise everything would be very tangled with the UserProfile case.

My suggestion was to implement something like this:
```
sealed class DialogState {

    abstract val title: String
    abstract val text: String
    abstract val status: UserStatus
    abstract var notShowAgain: Boolean

    data class StateAvailable(override var notShowAgain: Boolean = false) : DialogState() {
        override val title = stringResource(R.string.user_profile_change_status_dialog_available_title)
        override val text = stringResource(R.string.user_profile_change_status_dialog_available_text)
        override val status = UserStatus.AVAILABLE
    }

    data class StateBusy(override var notShowAgain: Boolean = false) : DialogState() {
        override val title = stringResource(R.string.user_profile_change_status_dialog_busy_title)
        override val text = stringResource(R.string.user_profile_change_status_dialog_busy_text)
        override val status = UserStatus.BUSY
    }

    data class StateAway(override var notShowAgain: Boolean = false) : DialogState() {
        override val title = stringResource(R.string.user_profile_change_status_dialog_away_title)
        override val text = stringResource(R.string.user_profile_change_status_dialog_away_text)
        override val status = UserStatus.AWAY
    }

    data class StateNone(override var notShowAgain: Boolean = false) : DialogState() {
        override val title = stringResource(R.string.user_profile_change_status_dialog_none_title)
        override val text = stringResource(R.string.user_profile_change_status_dialog_none_text)
        override val status = UserStatus.NONE
    }
}
```